Hestia/app/src/main/java/hestia/backend/Device.java

```diff
@@ -65,4 +65,27 @@ public void setActivators(ArrayList<Activator> activators) {
     public String toString(){
         return name +" "+ deviceId + " " + activators + "\n";
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Device device = (Device) o;
+
+        if (deviceId != device.deviceId) return false;
+        if (!name.equals(device.name)) return false;
```

Comment:
I think that there should be a special dialog for this in the client, since the client should decide on the name. At some point we should stop storing the device name at the server altogether, as it can be different for each client. The deviceID can then be the primary key.